radio/src/gui/common/stdlcd/radio_sdmanager.cpp

```diff
@@ -105,13 +105,14 @@ void onSdManagerMenu(const char * result)
     memcpy(reusableBuffer.sdmanager.originalName, line, sizeof(reusableBuffer.sdmanager.originalName));
     char * ext = getFileExtension(line, SD_SCREEN_FILE_LENGTH+1);
     if (ext) {
+      size_t len = strlen(ext);
```

Comment:
```diff
--- a/radio/src/gui/common/stdlcd/radio_sdmanager.cpp
+++ b/radio/src/gui/common/stdlcd/radio_sdmanager.cpp
@@ -103,18 +103,11 @@ void onSdManagerMenu(const char * result)
   }
   else if (result == STR_RENAME_FILE) {
     memcpy(reusableBuffer.sdmanager.originalName, line, sizeof(reusableBuffer.sdmanager.originalName));
-    char * ext = getFileExtension(line, SD_SCREEN_FILE_LENGTH+1);
-    if (ext) {
-      size_t len = strlen(ext);
-      // write spaces to allow a longer filename
-      memset(ext, ' ', SD_SCREEN_FILE_LENGTH-len-(ext-line));
-      line[SD_SCREEN_FILE_LENGTH-len] = '\0';
-    }
-    else {
-      size_t len = strlen(line);
-      memset(line + len, ' ', SD_SCREEN_FILE_LENGTH - len);
-      line[SD_SCREEN_FILE_LENGTH] = '\0';
-    }
+    uint8_t fnlen = 0, extlen = 0;
+    getFileExtension(line, 0, LEN_FILE_EXTENSION_MAX, &fnlen, &extlen);
+    // write spaces to allow extending the length of a filename
+    memset(line + fnlen - extlen, ' ', SD_SCREEN_FILE_LENGTH - fnlen + extlen);
+    line[SD_SCREEN_FILE_LENGTH /* - extlen ? */] = '\0';
     s_editMode = EDIT_MODIFY_STRING;
     editNameCursorPos = 0;
   }
```

I'm not sure why the `-len` in the original: `line[SD_SCREEN_FILE_LENGTH-len] = '\0';` ?  Seems like it should always be just SD_SCREEN_FILE_LENGTH to ensure null terminator... maximum name length is enforced later anyway in editName().